src/gamepad/gamepad-monitor.vala

```diff
@@ -0,0 +1,63 @@
+// This file is part of GNOME Games. License: GPLv3
+
+/**
+ * This class provides a way to the client to monitor gamepads
+ *
+ * The client interfaces with this class primarily
+ */
+public class LibGamepad.GamepadMonitor : Object {
+	/**
+	 * Emitted when a gamepad is plugged in
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_plugged (Gamepad gamepad);
+
+	/**
+	 * Emitted when a gamepad is unplugged
+	 * @param  gamepad    The gamepad
+	 */
+	public signal void gamepad_unplugged (Gamepad gamepad);
```

Comment:
This would be used when listing gamepads in a mappings dialog, plus it is logical for a monitor to emit plugged and unplugged signals